presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -602,6 +603,27 @@ public void createTable(Session session, String catalogName, ConnectorTableMetad
         metadata.createTable(session.toConnectorSession(connectorId), tableMetadata, ignoreExisting);
     }
 
+    @Override
+    public TableHandle createTemporaryTable(Session session, String catalogName, List<ColumnMetadata> columns, Optional<PartitioningMetadata> partitioningMetadata)
+    {
+        CatalogMetadata catalogMetadata = getCatalogMetadataForWrite(session, catalogName);
+        ConnectorId connectorId = catalogMetadata.getConnectorId();
+        ConnectorMetadata metadata = catalogMetadata.getMetadata();
+        ConnectorTableHandle connectorTableHandle = metadata.createTemporaryTable(
+                session.toConnectorSession(connectorId),
+                columns,
+                partitioningMetadata.map(partitioning -> createConnectorPartitioningMetadata(connectorId, partitioning)));
+        return new TableHandle(connectorId, connectorTableHandle);
+    }
+
+    private static ConnectorPartitioningMetadata createConnectorPartitioningMetadata(ConnectorId connectorId, PartitioningMetadata partitioningMetadata)
```

Comment:
The reason to extract this method seems to be able to use `partitioningMetadata.map(...)`. Although an `if-else` can also do this work? No strong opinion on this case though.

Reply:
I've tried to inline it and use the if statement, but it seems to be less readable, as you need to use `Optional#get` in multiple places, and you need to explicitly wrap the `ConnectorPartitioningMetadata` into Optional when handling it with an `if` statement. I would prefer to leave it as is. As it allows to use the fluent, functional style.